docs/evaluation/how_to_guides/evaluation/evaluate_with_attachments.mdx

```diff
@@ -0,0 +1,103 @@
+import {
+  CodeTabs,
+  python,
+  typescript,
+  PythonBlock,
+  TypeScriptBlock,
+} from "@site/src/components/InstructionsWithCode";
+
+# Evaluate an LLM application with attachments
+
+Attachments allow you to associate large files with your examples. This allows you to evaluate RAG applications
+over large internal documents, benchmark image analysis tools, etc.
+
+## Create a dataset with attachments
+
+To create a dataset with attachments, you need to use the `upsert_examples_multipart` method of the LangSmith client:
+
+```python
+from langsmith.client import Client
+from langsmith.schemas import ExampleUpsertWithAttachments
+
+# Pass in your api key directly, or define it in the LANGCHAIN_API_KEY environment variable
+langchain_client = Client(api_key="...")
+
+dataset = langchain_client.create_dataset(
+    dataset_name="attachment-test-dataset",
+    description="Test dataset for evals with attachments",
+)
+
+# Define the example
+example = ExampleUpsertWithAttachments(
+    dataset_id=dataset.id,
+    inputs={"question": "What were the cumulative earnings earned from online orders in the midwest during Q2?"},
+    outputs={"answer": "$123456"},
+    attachments={
+        # Each attachments is just a name with a mime type and the bytes content of the file
+        "pdf": ("application/pdf", Path("./foo_earnings.pdf").read_bytes()),
+        # We can pass multiple attachments (of different types!), as long as they have different names
+        "pptx": ("application/pptx", Path("./foo_earnings.pptx").read_bytes()),
+    },
+)
+
+# Upsert the examples
+langchain_client.upsert_examples_multipart(upserts=[example])
```

Comment:
ooc could this just be part of langchain_client.create_example? and we route to multipart upsert if attachments are passed in?

Reply:
not sure @agola11 might have thoughts here